src/compiler/checker.ts

```diff
@@ -4332,9 +4521,37 @@ namespace ts {
                 case SyntaxKind.QualifiedName:
                     let symbol = getSymbolAtLocation(node);
                     return symbol && getDeclaredTypeOfSymbol(symbol);
-                default:
+                case SyntaxKind.JSDocAllType:
+                    return anyType;
+                case SyntaxKind.JSDocUnknownType:
                     return unknownType;

```

Comment:
`unknownType` is a sentinel for the checker that we were unable to resolve an actual type at a location, and that we should fall back to `any`. I'd rather just say we're the `any` type since that's what we're bubbling up to the user anyhow.